lib/src/models/multireddit.dart

```diff
@@ -3,13 +3,406 @@
+    case IconName.cars:
+      return "cars";
+    case IconName.comic:
+      return "comics";
+    case IconName.cuteAnimals:
+      return "cute animals";
+    case IconName.diy:
+      return "diy";
+    case IconName.entertainment:
+      return "entertainment";
+    case IconName.foodAndDrink:
+      return "food and drink";
+    case IconName.funny:
+      return "funny";
+    case IconName.games:
+      return "games";
+    case IconName.grooming:
+      return "grooming";
+    case IconName.health:
+      return "health";
+    case IconName.lifeAdvice:
+      return "life advice";
+    case IconName.military:
+      return "military";
+    case IconName.modelsPinup:
+      return "models pinup";
+    case IconName.music:
+      return "music";
+    case IconName.news:
+      return "news";
+    case IconName.philosophy:
+      return "philosophy";
+    case IconName.picturesAndGifs:
+      return "pictures and gifs";
+    case IconName.science:
+      return "science";
+    case IconName.shopping:
+      return "shopping";
+    case IconName.sports:
+      return "sports";
+    case IconName.style:
+      return "style";
+    case IconName.tech:
+      return "tech";
+    case IconName.travel:
+      return "travel";
+    case IconName.unusualStories:
+      return "unusual stories";
+    case IconName.video:
+      return "video";
+    case IconName.emptyString:
+      return "";
+    case IconName.none:
+      return "None";
+    default:
+      throw new DRAWInternalError('IconName: $iconName is not supported');
+  }
+}
 
 /// A class which represents a Multireddit, which is a collection of
-/// [Subreddit]s. This is not yet implemented.
+/// [Subreddit]s.
+// TODO(@ckartik): Add CommentHelper.
 class Multireddit extends RedditBase with RedditBaseInitializedMixin {
+  static const String _kDisplayName = 'display_name';
+  static const String _kFrom = "from";
+  static const String _kMultiApi = 'multireddit_api';
+  static const String _kMultiredditRename = 'multireddit_rename';
+  static const String _kMultiredditUpdate = 'multireddit_update';
+  static const String _kSubreddits = "subreddits";
+  static const String _kTo = "to";
+  static const String _kVisibility = "visibility";
+  static const String _kWeightingScheme = "weighting_scheme";
+  static const int _redditorNameInPathIndex = 2;
+  static final _subredditRegExp = new RegExp(r'{subreddit}');
+  static final RegExp _userRegExp = new RegExp(r'{user}');
+  static final RegExp _multiredditRegExp = new RegExp(r'{multi}');
+
+  // TODO(@ckartik): Try to make the _data['key_color'] value null.
+  Color get keyColor => new HexColor(_data['key_color']);
+
+  /// When was this [Multireddit] created.
+  DateTime get createdUtc => new DateTime.fromMillisecondsSinceEpoch(
+      data['created_utc'].round() * 1000,
+      isUtc: true);
+
+  /// The [IconName] associated with this multireddit.
+  ///
+  /// Can be one of { artAndDesign, ask, books, business, cars, comic,
```

Comment:
This might appear weird when docs are generated. It might not make sense to list them all here and just refer to the `IconName` enum definition.

Reply:
Changed.